peps/pep-0817.rst

```diff
@@ -0,0 +1,2528 @@
+selector provided by the PyTorch team. This complexity represents a
+fundamental scalability issue with the current tag system.
+
+This problem is not unique to PyTorch. Projects like `JAX
+<https://docs.jax.dev/en/latest/>`_, `NumPy <https://numpy.org/>`_,
+`SciPy <https://scipy.org/>`_, `scikit-learn
+<https://github.com/scikit-learn/scikit-learn>`_ and many others in the
+scientific Python ecosystem face
+similar hurdles. The core issue is that wheel tags are not extensible
+enough to handle the combinatorial complexity of build options. Wheel
+variants can capture GPU compatibility in wheel metadata and select
+the correct wheel at install time.
+
+
+Current workarounds and their limitations
+-----------------------------------------
+
+Separate package indexes as variants
+''''''''''''''''''''''''''''''''''''
+
+Projects such as `PyTorch <https://pytorch.org/get-started/locally/>`__
+and `RAPIDS <https://docs.rapids.ai/install/#selector>`__
+currently distribute packages that approximate "variants" through
+separate package indexes with custom URLs. We will use the
+example of PyTorch, while the problem, the workarounds and the impact
+for users also apply to other packages.
+
+.. figure:: pep-0817/pytorch_variant_selector.png
+    :alt: A grid-based selector for PyTorch versions. Individual rows
+          provide the choice of PyTorch Build (stable or nightly),
+          operating system (Linux, Mac, Windows), package (Pip,
+          LibTorch, Source), language (Python, C++ / Java), and Compute
+          Platform (CUDA 12.6, CUDA 12.8, CUDA 13.0, ROCM 6.4, CPU).
+          Below these rows, the pip install command for the selected
+          variant is provided, utilizing the --index-url parameter.
+
+    The PyTorch install selector
+    (https://pytorch.org/get-started/locally/, captured 22-Aug-2025)
+
+PyTorch uses a combination of index URLs per accelerator type and local
+version segments as accelerator tag (such as ``+cu130``, ``+rocm6.4`` or
+``+cpu``) . Users need to first determine the correct index URL for
+their system, and add an index specifically for PyTorch.
+
+.. code:: bash
+
+    pip install torch --index-url https://download.pytorch.org/whl/cu129
+
+Tools need to implement special handling for the way PyTorch uses local
+version segments. These requirements break the pattern that packages
+are usually installed with. Problems with installing PyTorch
+are a very common point of user confusion. To quantify this, on
+2025-12-05, 552 out of 8136, or 6.8%, of issues on `uv's issue tracker
+<https://github.com/astral-sh/uv/issues/>`__ contained the term "torch".
+
+Wheel variants remove this special casing and make GPU and TPU packages
+work just as well as regular packages with native code. They also reduce
+the burden on the maintainers to run separate infrastructure and to find
+and use non-standard features such as local version segments present on
+an index.
```

Comment:
Would delete this para. The _Motivation_ section should establish why the PEP/change is needed, rather than going into the solution. Regardless I don't think this text adds much / it applies to all the workaround cases equally